pandas/_libs/parsers.pyx

```diff
@@ -535,18 +535,25 @@ cdef class TextReader:
             self.parser_start = 0
             self.header = []
         else:
-            if isinstance(header, list) and len(header):
-                # need to artifically skip the final line
-                # which is still a header line
-                header = list(header)
-                header.append(header[-1] + 1)
-
-                self.parser.header_start = header[0]
-                self.parser.header_end = header[-1]
-                self.parser.header = header[0]
-                self.parser_start = header[-1] + 1
-                self.has_mi_columns = 1
-                self.header = header
+            if isinstance(header, list):
+                if len(header) > 1:
```

Comment:
any way to make this slightly less duplicative

e.g. you can pul out
```
self.header  = header
```

after the ``if len(header) > 1:`` block (as its for both if and else.